packages/data-provider/src/utils/submissionUtils.ts

```diff
@@ -304,8 +323,60 @@ const utils = (dependencies: BaseDependencies) => {
 			};
 		},
 
-		removeEntityFromSubmission: (submissionData: Record<string, SubmissionInsertData>, entityName: string) => {
-			return _.omit(submissionData, entityName);
+		removeItemsFromSubmission: (
+			submissionData: SubmissionData,
+			filter: { actionType: SubmissionActionType; entityName: string; index: number | null },
```

Comment:
https://github.com/overture-stack/lyric/pull/93#discussion_r1725699061

Reply:
changed to `null` instead of `undefined` as we intentionally want to indicate that the variable `index` is set to "no value". Whereas setting to `undefined` could be interpreted the variable `index` has not been assigned a value yet.

Reply:
Responded to previous comment.